### PR TITLE
Rename to fontobene-qt & add Qt6 compatibility

### PR DESCRIPTION
**Pull request by ubruhin**
As described in https://github.com/fontobene/fontobene-qt5/issues/12#issuecomment-1971427580:

- Remove qmake support
- Rename to `fontobene-qt`, including include path & CMake targets
- Rename `fontobene-qt5.pc.example` to `fontobene-qt-5.pc.example` and add `fontobene-qt-6.pc.example`
- Add Qt6 compatibility
- Bump version to 1.0.0

~~Note: This is considered as draft but GitHub does not let me mark it as draft as long as it's a private repo~~ :see_no_evil: 

/cc @dbrgn

### CODE REVIEW

File: CHANGELOG.md
Comment (dbrgn) [inline CHANGELOG.md:1]: Why did you remove the Changelog? 🙂

Comment (ubruhin) [reply CHANGELOG.md:1]: Because it is basically a new library :wink: My intention is to add a new entry "changes compared to fontobene-qt5 0.2.0" when creating the first release, but I'd also be okay with keeping the complete changelog.

Comment (dbrgn) [reply CHANGELOG.md:1]: Ahh, I just realized that as well 😄👍️

File: README.md [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
 
 ## pkg-config
 
-If you're packaging fontobene-qt5 and want an example pkg-config file, check
-out `fontobene-qt5.pc.example`. Depending on your distro, the `includedir`
-might need adjusting.
+If you're packaging fontobene-qt and want an example pkg-config file, check
+out `fontobene-qt-qt5.pc.example` (for Qt5) resp. `fontobene-qt-qt6.pc.example`
+(for Qt6). Both should be provided to let users choose the Qt version.
+Depending on your distro, the `includedir` might need adjusting.
```
Comment (dbrgn) [inline README.md]: The word "respectively" is used a bit differently in English than in German: https://www.linguee.de/deutsch-englisch/search?source=auto&query=respectively

I'd suggest the following:

```suggestion
If you're packaging fontobene-qt and want an example pkg-config file, check
out `fontobene-qt-qt5.pc.example` (for Qt5) or `fontobene-qt-qt6.pc.example`
(for Qt6). Both should be provided to let users choose the Qt version.
Depending on your distro, the `includedir` might need adjusting.
```

File: fontobene-qt-qt6.pc.example [hunk 2]
```diff
@@ -0,0 +1,9 @@
+prefix=/usr
+includedir=${prefix}/include
+
+Name: fontobene-qt-qt6
```
Comment (dbrgn) [inline fontobene-qt-qt6.pc.example]: Do we need to name the versioned variants `fontobene-qt-qt6`, or could we use `fontobene-qt6`? I assume this is due to the collision potential with the `fontobene-qt5` codebase? If yes, would that really be a problem? (In some way, `fontobene-qt` replaces `fontobene-qt5` and increments the version, so it should be fine.)

**Review by dbrgn — APPROVED** [review]

File: fontobene-qt-qt6.pc.example (hunk 2 above)
Comment (ubruhin) [reply fontobene-qt-qt6.pc.example]: Hmm so you mean only the name written in the *.pc file? :thinking: 

Actually I named it just identical to the filename. But I don't know exactly what it is used for. I just found [this](https://people.freedesktop.org/~dbn/pkg-config-guide.html):

> Name: A human-readable name for the library or package. This does not affect usage of the pkg-config tool, which uses the name of the .pc file.

So I suspect it is not really relevant and I'd agree removing the redundant `-qt` :+1: 

However, for the filename itself [I suspect](https://github.com/fontobene/fontobene-qt5/issues/12#issuecomment-1971427580) it's better to avoid collision with `fontobene-qt5`(?)

Comment (ubruhin) [reply fontobene-qt-qt6.pc.example]: Btw, for the filename we might consider dropping the additional `qt` while still keeping the dash for separation. The filenames would then be:

```
# fontobene-qt5
fontobene-qt5.pc

# fontobene-qt
fontobene-qt-5.pc
fontobene-qt-6.pc
```

Not very elegant, but no duplicate `qt` while still avoiding the conflict...

Comment (dbrgn) [reply fontobene-qt-qt6.pc.example]: I think that looks better, but I have no strong opinion on it 🙂

Comment (ubruhin) [reply fontobene-qt-qt6.pc.example]: Alright, renamed the *.pc files and set "Name: fontobene-qt" for both files. I think now this PR should be ready to merge.
